src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,131 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.CurrentLoginIsExistsException;
+import core.basesyntax.model.InvalidInputDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String DEFAULT_LOGIN = "programator12341";
+    private static final String DEFAULT_PASSWORD = "passatic123";
+    private static final int DEFAULT_AGE = 19;
+    private static final String SHORT_PASSWORD = "short";
+    private static final String SHORT_LOGIN = "short";
+    private static final String MISMATCH_LOGIN = "--------------";
+    private static final String MISMATCH_PASSWORD = "%$#%@**(*&*(";
+
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private final StorageDao storageDao = new StorageDaoImpl();
+    private final User defaultUser = new User();
+    private final User defaultCopyUser = new User();
+    private final User nullUser = new User();
+    private final User nullPasswordUser = new User();
+    private final User nullLoginUser = new User();
+    private final User nullAgeUser = new User();
+    private final User shortPasswordUser = new User();
+    private final User shortLoginUser = new User();
+    private final User misMathcPatternUser = new User();
+
+    @BeforeEach
+    void setUp() {
+        Storage storage = new Storage();
+        storage.people.clear();
+        defaultUser.setAge(18);
+        defaultUser.setLogin(DEFAULT_LOGIN);
+        defaultUser.setPassword(DEFAULT_PASSWORD);
```

Comment:
just setup here the one user with all default values, and change needed values in the test methods below